src/structs.rs

```diff
@@ -61,132 +61,120 @@ impl Keyword {
-                "Cannot create a block holding more than {} block",
-                LENGTH - 1
-            )));
-        }
-        // The default pads the entire block with 0s.
-        let mut block = Self::default();
-        if is_terminating {
-            block[0] = bytes.len() as u8;
-        } else {
-            // No block can contain `LENGTH` bytes.
-            block[0] = LENGTH as u8;
-        }
-        for (i, b) in bytes.iter().enumerate() {
-            block[i + 1] = *b;
-        }
-        Ok(block)
-    }
-
-    /// Unpads the byte vectors contained in the given list of `Block`.
-    ///
-    /// - `blocks`  : list of blocks to unpad
-    pub fn unpad(blocks: &[Self]) -> Result<Vec<Vec<u8>>, Error> {
-        let mut blocks = blocks;
-        let mut res = Vec::new();
-        while !blocks.is_empty() {
-            // At least `LENGTH - 1` bytes will be read
-            let mut bytes = Vec::with_capacity(LENGTH - 1);
-            while blocks.len() > 1 && blocks[0][0] == LENGTH as u8 {
-                bytes.extend_from_slice(&blocks[0][1..]);
-                blocks = &blocks[1..];
-            }
-            let length = blocks[0][0] as usize;
-            if length == LENGTH {
-                return Err(Error::CryptoError(
-                    "Last block given is a non-terminating block.".to_string(),
-                ));
-            }
-            bytes.extend_from_slice(&blocks[0][1..=length]);
-            blocks = &blocks[1..];
-            res.push(bytes);
-        }
-        Ok(res)
-    }
-
-    /// Pads the given bytes into blocks. Uses the first byte to differentiate
-    /// terminating blocks:
-    ///
-    /// - non-terminating blocks are prefixed by `LENGTH`;
-    /// - terminating blocks are prefixed by the actual length written.
-    ///
-    /// # Parameters
-    ///
-    /// - `bytes`   : bytes to be padded in to a `Block`
-    pub fn pad(mut bytes: &[u8]) -> Result<Vec<Self>, Error> {
-        // Number of blocks needed.
-        let mut partition_size = bytes.len() / (LENGTH - 1);
-        if bytes.len() % (LENGTH - 1) != 0 {
-            partition_size += 1;
-        }
-
-        let mut blocks = Vec::with_capacity(partition_size);
-        while bytes.len() > LENGTH - 1 {
-            // this is a non-terminating block
-            blocks.push(Self::new(&bytes[..LENGTH - 1], false)?);
-            bytes = &bytes[LENGTH - 1..];
+        let mut data = [0; LENGTH];
+        data[..bytes.len()].copy_from_slice(bytes);
+        Ok(Self {
+            block_type,
+            prefix,
+            data,
+        })
+    }
+
+    /// Returns `true` if this block is an addition, and `false` if it is a
+    /// deletion.
+    pub fn is_addition(&self) -> bool {
+        self.block_type == BlockType::Addition
+    }
+
+    /// Generates a new padding block.
+    pub const fn padding_block() -> Self {
+        Self {
+            block_type: BlockType::Deletion,
```

Comment:
Why padding blocks are `BlockType::Deletion`? In my understanding it doesn't matter but :
1. Is it hard to have `block_type` info only on not padding block?
2. If it is, can you add a comment explaining that this value is meaningless?

Reply:
done in f2c5ec6